server/chat-formatter.js

```diff
@@ -222,7 +222,7 @@ class TextFormatter {
 	 * @param {string} html
 	 */
 	toUriComponent(html) {
-		let component = html.replace(/&amp;/g, '&').replace(/&lt;/g, '<').replace(/&gt;/g, '>').replace(/&quot;/g, '"').replace(/&apos;/g, '\'');
+		let component = html.replace(/&lt;/g, '<').replace(/&gt;/g, '>').replace(/&quot;/g, '"').replace(/&apos;/g, '\'');
```

Comment:
was it wanted behaviour that a chat message of `&gt;` became `&amp;gt;` and subsequently `>` due to the uri encode? if not this should be fixed now

Reply:
You're right, that was unintentional. The `&amp;` replacement should go at the end.

```suggestion
		let component = html.replace(/&lt;/g, '<').replace(/&gt;/g, '>').replace(/&quot;/g, '"').replace(/&apos;/g, '\'').replace(/&amp;/g, '\'');
```

This code is intended to turn a chat message of `[[wiki: D&D]]` into a link to `//en.wikipedia.org/w/index.php?title=Special:Search&search=D%26D` – I can add that to unit tests to make sure it's not wrong.

Reply:
I'll add tests because it's hard to tell if this code is correct.

Reply:
This doesn't fix the issue I described above. uriencode already encodes `&amp;` into `&`, manually replacing it beforehand will always lead to that issue, so it should not be replaced manually at all.

Reply:
`encodeURIComponent('&amp;')`
=> `"%26amp%3B"`

I'm not getting that result, but maybe I'm misinterpreting. Either way, I'll write unit tests and maybe I'll figure it out there. If we're still not in consensus after that, you can write the unit test that you think is wrong.

Reply:
I've added tests and I don't see the problem you're trying to point out. If you still think I'm wrong, can give a testcase?

Reply:
Just checked and encodeURI and encodeURIComponent work differently, so you're correct